Mlem/Models/Content/Community/CommunityModel.swift

```diff
@@ -0,0 +1,170 @@
+//
+//  CommunityModel.swift
+//  Mlem
+//
+//  Created by Sjmarf on 20/09/2023.
+//
+
+import Dependencies
+import Foundation
+
+struct CommunityModel {
+    @Dependency(\.apiClient) private var apiClient
+    @Dependency(\.errorHandler) var errorHandler
+    @Dependency(\.hapticManager) var hapticManager
+    @Dependency(\.communityRepository) var communityRepository
+    
+    enum CommunityError: Error {
+        case noData
+    }
+    
+    @available(*, deprecated, message: "Use attributes of the CommunityModel directly instead.")
+    var community: APICommunity
+    
+    // Ids
+    let communityId: Int
+    let instanceId: Int
+    
+    // Text
+    let name: String
+    let displayName: String
+    let description: String?
+    
+    // Images
+    let avatar: URL?
+    let banner: URL?
+    
+    // State
+    var nsfw: Bool
+    var local: Bool
+    var removed: Bool
+    var deleted: Bool
+    var hidden: Bool
+    var postingRestrictedToMods: Bool
+    
+    // From APICommunityView
+    var blocked: Bool?
+    var subscribed: Bool?
+    var subscriberCount: Int?
```

Comment:
These values get nil-checked a lot but don't have meaningful nil states--probably nicer to remove the optional and default a `nil` from the API model to false/0

Reply:
Hmm, I'm not sure what to do about this one. How do we know if a value is loaded or not? Setting `subscribed` to `false` when we don't actually know whether the user is subscribed or not would cut down on the amount of null coalescing we have to do View-side, but could be misleading.

Notably, I'm doing a check to test if one of these values is `nil` in `SidebarView`. If the value is `nil`, it will fetch the extra information. If we make these values non-optional, we'd have to add a `hasLoaded` property or something.

Reply:
Ah, yes, that's a good point--I didn't catch that these also store loading state. We can leave as-is for now.